storage/series/reader.go

```diff
@@ -210,19 +214,10 @@ func (r Reader) fetchBlocksWithBlocksMapAndBuffer(
 		case r.retriever != nil:
 			// Try to stream from disk
 			if r.retriever.IsBlockRetrievable(start) {
-				if clonedID == nil {
-					// Clone ID as the block retriever uses the ID async so we cannot
-					// be sure about owner not finalizing the cloneableID passed to
-					// the Reader
-					clonedID = r.opts.IdentifierPool().Clone(r.cloneableID)
-					ctx.RegisterFinalizer(clonedID)
-				}
-				// NB(r): Always use nil for onRetrieveBlock so we don't cache the
-				// series after fetching it from disk, the fetch blocks API is called
-				// during streaming so to cache it in memory would mean we would
-				// eventually cache all series in memory when we stream results to a
-				// peer.
-				var onRetrieve block.OnRetrieveBlock
+				// Clone ID as the block retriever uses the ID async so we cannot
+				// be sure about owner not finalizing the cloneableID passed to
+				// the Reader
+				clonedID := r.opts.IdentifierPool().Clone(r.cloneableID)
```

Comment:
Regarding my previous question, you're able to remove the RegisterFinalizer() call because the onRetrieve method that caches the data will actually finalize it right?

Reply:
Actually you made sure that onRetrieve was nil, so won't this leak the clonedID?

Reply:
Yeah good catch, for FetchBlocks it might actually make sense to continue using `ctx.RegisterFinalizer(...)` for cleaning up the IDs.